### PR TITLE
dm: support config experimental features

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve?

* support config experimental features

### What is changed and how does it work?

add new struct `DMExperimental` in `MasterConfig`

### Code changes

- [x] Has Go code change
- [x] Has CI related scripts change

### Tests

- [x] Unit test 
- [x] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
None
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/dm_config.go [hunk 1]
```diff
@@ -55,6 +55,11 @@ type MasterConfig struct {
 	// +optional
 	// +k8s:openapi-gen=false
 	DMSecurityConfig `toml:",inline" json:",inline"`
+
+	// dm-master's experimental config
+	// +optional
+	// +k8s:openapi-gen=false
+	DMExperimental *DMExperimental `toml:"experimental,omitempty" json:"experimental,omitempty"`
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/dm_config.go:62]:
```suggestion
	Experimental *DMExperimental `toml:"experimental,omitempty" json:"experimental,omitempty"`
```

Comment (DanielZhangQD) [reply pkg/apis/pingcap/v1alpha1/dm_config.go:62]: Could you please show the toml configurations in DM?

Comment (DanielZhangQD) [reply pkg/apis/pingcap/v1alpha1/dm_config.go:62]: No need to add this for the dm-worker?

File: pkg/apis/pingcap/v1alpha1/dm_config.go [hunk 2]
```diff
@@ -95,3 +100,8 @@ type DMSecurityConfig struct {
 	// +optional
 	CertAllowedCN []string `toml:"cert-allowed-cn,omitempty" json:"cert-allowed-cn,omitempty" yaml:"cert-allowed-cn,omitempty"`
 }
+
+// DM experimental config
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/dm_config.go:104]:
```suggestion
// DM experimental config
// +k8s:openapi-gen=true
```

Comment (Ehco1996) [reply pkg/apis/pingcap/v1alpha1/dm_config.go:104]: thanks for  review, now dm-master config looks like [this](https://github.com/pingcap/ticdc/blob/9721053edae6b146b726bd5daec04a0dc8c57b1a/dm/dm/master/dm-master.toml#L46)

and this new config only worked in dm-master not affects dm-worker

File: tools/tools.go [hunk 3]
```diff
@@ -11,6 +11,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//go:build tools
```
Comment (DanielZhangQD) [inline tools/tools.go:14]: This should be removed.

**Comment by Ehco1996** [comment]
seems e2e test of DM is passed but other test failed, so i think this pr is ready for review

ptal  @DanielZhangQD @csuzhangxc  🧡

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Comment by DanielZhangQD** [comment]
@Ehco1996 e2e failed https://ci.pingcap.net/blue/organizations/jenkins/tidb-operator-pull-e2e-kind/detail/tidb-operator-pull-e2e-kind/1628/tests.

**Comment by Ehco1996** [comment]
> @Ehco1996 e2e failed https://ci.pingcap.net/blue/organizations/jenkins/tidb-operator-pull-e2e-kind/detail/tidb-operator-pull-e2e-kind/1628/tests.

>   /bin/sh: curl: not found
how should i fix this? i am not sure if install curl into pod is a good way to solve this

File: pkg/apis/pingcap/v1alpha1/dm_config.go [hunk 4]
```diff
@@ -95,3 +99,9 @@ type DMSecurityConfig struct {
...
+// +k8s:openapi-gen=true
+type DMExperimental struct {
+	OpenAPI bool `toml:"openapi,omitempty" json:"openapi,omitempty" yaml:"openapi,omitempty"`
```
Comment (csuzhangxc) [inline pkg/apis/pingcap/v1alpha1/dm_config.go:106]: How about adding a comment for which version of DM this feature was introduced?

File: pkg/apis/pingcap/v1alpha1/dm_config.go [hunk 5]
```diff
@@ -103,5 +103,6 @@ type DMSecurityConfig struct {
 // DM experimental config
 // +k8s:openapi-gen=true
 type DMExperimental struct {
+	// OpenAPI was introduced in DM V5.3.0
```
Comment (csuzhangxc) [inline pkg/apis/pingcap/v1alpha1/dm_config.go:106]: you need to re-generate CRD and some other code.

**Review by csuzhangxc — APPROVED** [review]

**Comment by Ehco1996** [comment]
/[Fail] TiDBCluster [Feature: Upgrade] while concurrently scale TiKV

/test pull-e2e-kind
/test pull-e2e-kind-serial

**Comment by Ehco1996** [comment]
[Fail] TiDBCluster [Feature: Upgrade] while concurrently scale PD [It] out 😅

**Comment by Ehco1996** [comment]
[Fail] TiDBCluster [Feature: CDC] [It] should scale in-out and mount storage volumes correctly

**Pull request merged**